### ISSUE TITLE
[Messenger] Not working well together with autoconfigure attribute

### ISSUE THREAD

**Issue by icanhazstring** [symfony/symfony#43175]
**Symfony version(s) affected**: 5.3.*

**Description**
Trying to only configure a single MessageHandler using `AutoconfigureTag` in conjunction with `MessageHandlerInterface` leads to double registered service. 

**How to reproduce**
Writing a Handler for any type of event using the `MessageHandlerInterface` will result in that service to be registered as message handler when running `bin/console debug:messenger`

Trying to add the `from_transport` tag using the `AutoconfigureTag` results in two handlers to be registered. This will run two handlers for the same message when its coming from that configured transport. 

The solution is to add this into services.yaml

```
App\Habdler:
  autoconfigure: false
  tags:
    { name: 'messenger.message_handler' , from_transport: '...'} 
```

This results in only a single handler. 
It is not possible to achieve this using `AutoconfigureTag` only, but I think it should. 

**Possible Solution**
If you point me to the source where to solve this I am happy to provide a PR.

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by icanhazstring** [comment]
Will do 👍

**Comment by icanhazstring** [comment]
@xabbuh created a small symfony project displaying this problem.
Also added a readme with the problem.

See: https://github.com/icanhazstring/symfony-messenger-43175

**Comment by icanhazstring** [comment]
I guess this is fixed with symfony 5.4?
https://symfony.com/blog/new-in-symfony-5-4-messenger-improvements#configurable-handlers-with-php-attributes

**Comment by xabbuh** [comment]
Let's close here then. Thank you for the feedback.
